src/blocks/cpu.rs

```diff
@@ -239,3 +242,33 @@ impl Block for Cpu {
         &self.id
     }
 }
+
+#[inline]
+fn format_utilization(values: &[f64], count: usize, per_core: bool) -> String {
+    if per_core {
+        values
+            .iter()
+            .take(count)
+            .skip(1) // The first value is a global one.
+            .map(|v| format!("{:02.0}%", 100.0 * v))
+            .collect::<Vec<String>>()
+            .join(" ")
+    } else {
+        format!("{:02.0}", 100.0 * values[0])
+    }
+}
+
+#[inline]
+fn format_frequency(cpu_freqs: &[f32], count: usize, per_core: bool) -> String {
+    if per_core {
+        cpu_freqs
+            .iter()
+            .take(count)
+            .map(|v| format!("{0:.1}GHz", v / 1000.0))
```

Comment:
Is there a reason only `per_core` displayed "Ghz"?

Reply:
From what I understood, the current code wasn't adding the "unit": it's up to the user to add it in the format string (as in `format = "{barchart} {utilization}% {frequency}GHz" ` from the documentation).

Now, with N values I find it clearer to add the unit to each value, but you're right: that's no longer consistent with the single-value case.

Maybe I can also add the "unit" in the single-value case and update the example format in the doc (and the default one in the code) to remove the `%`, what do you think?

Reply:
I've added the unit even when `per_core=false`, to be consistent.

But as mentioned in the commit message, I think it may surprise some users when they update, if they already added %/GHz in their format string (maybe we can mention it in the changelog though).

Anyway, it's in a dedicated commit so I can easily remove/rework it if need be :slightly_smiling_face:

Reply:
I'm also concerned about breaking existing configurations here.

Reply:
Yeah, that can be an issue and that's why I only added the "unit" when `per_core` is `true` (it's `false` by default, so no breakage on upgrade), but as pointed out it's a bit surprising when reading the code/inconsistent.

Possible solutions:
- add a comment explaining why we don't add unit in the else branch (backward compat)
- don't add unit even with `per_core=true` (but I think the result is less legible)
- add another flag, `add_unit` (may be overkill, and I think we don't want to multiply the number of boolean flags)
- something I haven't thought of?

Reply:
@ammgws or @atheriel : Anything I could do to make this move forward :slightly_smiling_face:?
Have you a preferred option/solution?

Reply:
Still thinking about whether there's another way

Reply:
Perhaps revert the last commit for now

Reply:
Done :slightly_smiling_face:

Reply:
I think we can figure this out later